components/RadioPlayer.vue

```diff
@@ -255,34 +264,20 @@ export default {
       }
     },
     show_url () {
-      const arcsiShowsList = [...this.arcsiShows]
-      let this_show
-      if (this.np.live.is_live) // live show
-      { this_show = arcsiShowsList.find(show => show.name === this.np.live.streamer_name) } else // pre-recorded show
-      { this_show = arcsiShowsList.find(show => show.name === this.np.now_playing.song.artist) }
-      let url = ''
-      try {
-        url = this_show.archive_lahmastore_base_url
-      } catch (error) { // Happens on the first call after page is loaded, not sure why...
-        // console.log("No show URL found.";
-      }
+      const url = this.currentShowArcsi ? this.currentShowArcsi.archive_lahmastore_base_url : ''
       return '/shows/' + url
     },
     show_art_url () {
       if (this.np.live.is_live) {
-        const tryArtFromShow = this.showsList_lookup[this.np.live.streamer_name] // try to find show artwork url based on streamer name
-        if (tryArtFromShow === undefined) { // show not found
-          return this.default_art_url
-        } // return default
-        else { return tryArtFromShow } // resturn show art work
+        // check if this method is valid with streams
+        return this.currentShowArcsi ? this.currentShowArcsi.cover_image_url : this.default_art_url
       } else {
         const songTitleJSON = this.np.now_playing.song.title
         const songArtistJSON = this.np.now_playing.song.artist
         const artworkJSON = this.np.now_playing.song.art // art work url in json
 
         if (artworkJSON === this.default_azuracast_art_url) { // default url by azuracast (must be returning off air music with art work)
-          const tryArtFromShow = this.showsList_lookup[songArtistJSON] // try to find show artwork url based on show title
-          if (tryArtFromShow === undefined) { // show not found
+          if (this.currentShowArcsi.cover_image_url === undefined) { // show not found
```

Comment:
Let's get rid of searching through the history, no? I don't think it's used anyways. 

Plus, is it a real threat that `this.currentShowArcsi.cover_image_url` is undefined? It can be an assumption on arcsi's side, no?

Reply:
I don't know why it was checked, but it was in the old script, maybe there were some cases like that? I don't think it is bad to check for it, if it's ok, it will just pass it, no?